autoload/pandoc/completion.vim

```diff
@@ -18,8 +18,68 @@ function! pandoc#completion#Init() abort "{{{1
             let g:pandoc#completion#bib#use_preview = 0
         endif
     endif
-
+    "
+    " If your images are in the same directory as the markdown file, they are included
+    " by default in the completion menu. However, many authors like to keep their
+    " images in a directory such as 'images' or 'figures'. The options called
+    " 'g:pandoc#completion#imgdirtype' and 'g:pandoc#completion#imgdirpre' are
+    " used to define such image directories
+    "
+    let s:imgdir = ''
+    if exists('g:pandoc#completion#imgdirtype')
+        if g:pandoc#completion#imgdirtype == 1
+            if exists('g:pandoc#completion#imgdirpre')
+                let s:imgdir = g:pandoc#completion#imgdirpre
+            endif
+        elseif g:pandoc#completion#imgdirtype == 2
+            if exists('g:pandoc#completion#imgdirpre')
+                let s:imgdir = g:pandoc#completion#imgdirpre . expand('%:r')
+            endif
+        endif
+    endif
+    "
+    " User can specify the sources from where reference list consisting of
+    " labels such as fig:, lst:, eq: etc. will be populated. Currently two
+    " sourses are supported: g:pandoc#completion#refsources = 'buffers' and
+    " g:pandoc#completion#refsources = 'pandocfiles'. These correspond to all
+    " open buffers ('buffers') and the pandoc files found (recursively) in the
+    " current directory ('pandocfiles'), respectively. 'buffers' is the
+    " default
+    "
+    let s:refsources = 'buffers'
+    if exists('g:pandoc#completion#refsources')
+        let s:refsources = g:pandoc#completion#refsources
+    endif
+    "
+    let l:currArgs = argv()
+    execute '%argd'
+    if s:refsources ==# 'buffers'
+        " do nothing
+    elseif s:refsources ==# 'pandocfiles'
+        " this will populate the bufferlist and we can then work with bufdo
+        " argdo does not work very well with unsaved buffers
+        let l:pandocfiles = []
+        call extend(l:pandocfiles, glob('**/*.pandoc', 1, 1))
+        call extend(l:pandocfiles, glob('**/*.pdk', 1, 1))
+        call extend(l:pandocfiles, glob('**/*.pd', 1, 1))
+        call extend(l:pandocfiles, glob('**/*.pdc', 1, 1))
+        execute 'argadd ' . join(l:pandocfiles)
+        if get(g:, 'pandoc#filetypes#pandoc_markdown', 1) == 1
```

Comment:
For some reason the entries in globbed files in the current work dir did not appear.  Only those of open buffers.